static/js/composebox_typeahead.js

```diff
@@ -75,6 +75,8 @@ function query_matches_stream(query, stream) {
 
 // Case-insensitive
 function query_matches_emoji(query, emoji) {
+    // replaces spaces with underscores
+    query = query.split(" ").join("_");
```

Comment:
Just FYI, you can use a regex with g modifier to perform a global search but anyways that method will be less performant than the split-join method you have used. :)